### ISSUE TITLE
[BUG]: type OAuthAppAuthentication does not conform to OIDC specifications

### ISSUE THREAD

**Issue by ataylorme** [octokit/oauth-methods.js#279]
### What happened?

I am using [auth-oauth-user.js](https://github.com/octokit/auth-oauth-user.js/) to [generate a user access token using the web flow](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-user-access-token-for-a-github-app#using-the-web-application-flow-to-generate-a-user-access-token).

The responses are used in an OpenID Connect (OIDC) identity provider. The return type `OAuthAppAuthentication` does not conform to [OIDC specifications](https://datatracker.ietf.org/doc/html/rfc8693#section-2.2.1).

This is causing me to manually remap the returned value from `auth-oauth-user.js` from camelCase back to snake_case as originally provided by `https://github.com/login/oauth/access_token`

I believe `auth-oauth-user.js` should **not** alter the response from `https://github.com/login/oauth/access_token` and instead return snake_case

### Versions

"@octokit/auth-oauth-user": "^5.1.1"

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wolfy1339** [comment]
I agree with the sentiment, this will be a breaking change though as it changes the behaviour of the library
